python/tskit/tables.py

```diff
@@ -1505,6 +1505,17 @@ def __banner(self, title):
         title_line += "#"
         return line + "\n" + title_line + "\n" + line + "\n"
 
+    def __iter__(self):
+        """
+        Iterate over the tables in this TableCollection, ordered by table name
+        (i.e. deterministically), returning a tuple of (table_name, table_object)
+        """
```

Comment:
Or, 
```python
yield "edges", self.edges
yield "nodes", self.nodes
...
```

to avoid the ``getattrs`` (which are surprisingly slow)?

Reply:
I was trying to avoid actually hand-coding all the tables, in case new tables were added and we forgot to update this particular function. But happy with whatever strikes the correct balance between maintainability and speed.

Reply:
Not planning on adding extra tables, and if we did, this is the absolute least of the problems we'd need to deal with! I bet the total number of characters in the simpler version is about the same as what you have currently anyway.

Using the current version in the tests ensures that an error will be thrown if we do add tables and forget to update this anyway, so it's good to have the automatic version as well.